File: frontend/src/modules/enterprise/types/enterprise.ts
export interface CompliancePolicyItem {
  id: string;
  name: string;
  framework: string;
  status: string;
  lastAuditAt: string;
}

export interface PartnerItem {
  id: string;
  name: string;
  tier: string;
  commissionPct: number;
  status: string;
  createdAt: string;
}

export interface SupportTicketItem {
  id: string;
  subject: string;
  priority: string;
  slaStatus: string;
  status: string;
  createdAt: string;
}

export interface CustomerHealthItem {
  id: string;
  orgId: string;
  healthScore: number;
  churnRisk: string;
  npsScore: number;
  updatedAt: string;
}

export interface PlatformStatusItem {
  globalHealth: string;
  multiRegionClusters: number;
  activeMonitoredTenants: number;
  complianceScore: number;
  platformVersion: string;
}

File: backend/app/schemas/enterprise.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class CompliancePolicyResponse(BaseModel):
    id: str
    name: str
    framework: str
    status: str
    last_audit_at: datetime

    class Config:
        from_attributes = True

class PolicyCreateRequest(BaseModel):
    name: str
    framework: str = "SOC 2 Type II"
    policy_rules_json: Optional[str] = "{}"

class PartnerResponse(BaseModel):
    id: str
    name: str
    tier: str
    commission_pct: float
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

class PartnerCreateRequest(BaseModel):
    name: str
    tier: str = "Platinum Partner"
    commission_pct: float = 20.0

class SupportTicketResponse(BaseModel):
    id: str
    subject: str
    priority: str
    sla_status: str
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

class TicketCreateRequest(BaseModel):
    subject: str
    priority: str = "P1 - Critical"

class CustomerHealthResponse(BaseModel):
    id: str
    org_id: str
    health_score: float
    churn_risk: str
    nps_score: int
    updated_at: datetime

    class Config:
        from_attributes = True

class PlatformStatusResponse(BaseModel):
    global_health: str
    multi_region_clusters: int
    active_monitored_tenants: int
    compliance_score: float
    platform_version: str
